common/djangoapps/student/views.py

```diff
@@ -2271,3 +2272,34 @@ def change_email_settings(request):
         track.views.server_track(request, "change-email-settings", {"receive_emails": "no", "course": course_id}, page='dashboard')
 
     return JsonResponse({"success": True})
+
+
+def _get_course_programs(user, user_enrolled_courses): # pylint: disable=invalid-name
+    """ Return a list of course key containing data required
+    for dashboard.
+
+    Given a user and an iterable of course keys, find all
+    the programs relevant to the user's dashboard and return them in a
+    dictionary keyed by the course_key.
+
+    Arguments:
+        user (user object): Currently logged-in User
+        course_keys (list): List of course keys in which user is enrolled
+
+    Returns:
+        Dictionary response containing programs or {}
+    """
+    course_programs = get_course_programs_for_dashboard(user, user_enrolled_courses)
+    programs_data = {}
+    for course_key, program in course_programs.items():
+        if program.get('status') == 'active' and program.get('category') == 'xseries':
+            programs_data[course_key] = {
+                'course_count': len(program.get('course_codes', [])),
+                'display_name': program.get('name', ''),
+                'category': program.get('category'),
+                'program_marketing_url': urljoin(
+                    settings.MKTG_URLS.get('ROOT'), program.get('category') + '/{}').format(
```

Comment:
might as well just use literal 'xseries' here, since we won't be in this loop otherwise.